.github/workflows/ci.yml

```diff
@@ -3,8 +3,6 @@ name: CI
 # Controls when the action will run.
 on:
   # Triggers the workflow on push or pull request events but only for the main branch
-  push:
-    branches: [ main ]
   pull_request:
```

Comment:
This CI workflow is crazy. So it uses a runner to spin up an ubuntu container (?/something), installs python and ansible, then downloads the role using the requirements file, then runs lint on it?

And it would do this only when you .. complete (?) ... a pull request? Or runs before merging into Main?

And I assume the reason it has all of those depedencies listed is that after this completes, it just drops that image/container and rebuilds when it needs to run this workflow again